From e07120ef2aa671b5d4dafa37d54d6155f9cd5098 Mon Sep 17 00:00:00 2001
From: Daniele Ricci <daniele@casaricci.it>
Date: Sun, 15 Oct 2023 20:26:51 +0200
Subject: [PATCH 1/5] pipeline: strip XML tags from SSML speech text

---
 .../components/assist_pipeline/default_pipeline.py    | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/homeassistant/components/assist_pipeline/default_pipeline.py b/homeassistant/components/assist_pipeline/default_pipeline.py
index 25cbb9817bb..5d57b1827b3 100644
--- a/homeassistant/components/assist_pipeline/default_pipeline.py
+++ b/homeassistant/components/assist_pipeline/default_pipeline.py
@@ -835,6 +835,17 @@ class _DefaultPipelineProcessor:
 
         _LOGGER.debug("conversation result %s", conversation_result)
 
+        speech: str = conversation_result.response.speech.get("plain", {}).get(
+            "speech", ""
+        )
+        import xml.etree.ElementTree
+        try:
+            conversation_result.response.speech["plain"]["speech"] = ''.join(xml.etree.ElementTree
+                                                                             .fromstring(speech).itertext())
+        except xml.etree.ElementTree.ParseError:
+            # probably not a SSML message - just skip
+            pass
+
         self.host.process_event(
             PipelineEvent(
                 PipelineEventType.INTENT_END,
-- 
2.53.0

