From c3b2aebd10afd584f7b1c2cf9b800d2bd9d1a9f9 Mon Sep 17 00:00:00 2001
From: Daniele Ricci <daniele@casaricci.it>
Date: Sun, 15 Oct 2023 20:26:51 +0200
Subject: [PATCH 1/5] pipeline: strip XML tags from SSML speech text

---
 homeassistant/components/assist_pipeline/pipeline.py | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/homeassistant/components/assist_pipeline/pipeline.py b/homeassistant/components/assist_pipeline/pipeline.py
index a54bec88a86..323aa99a2ce 100644
--- a/homeassistant/components/assist_pipeline/pipeline.py
+++ b/homeassistant/components/assist_pipeline/pipeline.py
@@ -1333,6 +1333,17 @@ class PipelineRun:
 
         _LOGGER.debug("conversation result %s", conversation_result)
 
+        speech: str = conversation_result.response.speech.get("plain", {}).get(
+            "speech", ""
+        )
+        import xml.etree.ElementTree
+        try:
+            conversation_result.response.speech["plain"]["speech"] = ''.join(xml.etree.ElementTree
+                                                                             .fromstring(speech).itertext())
+        except xml.etree.ElementTree.ParseError:
+            # probably not a SSML message - just skip
+            pass
+
         self.process_event(
             PipelineEvent(
                 PipelineEventType.INTENT_END,
-- 
2.53.0

