From 33c77596f5501a5094ce0db695f71f75416826a7 Mon Sep 17 00:00:00 2001
From: Daniele Ricci <daniele@casaricci.it>
Date: Sun, 9 Aug 2026 21:40:12 +0200
Subject: [PATCH 5/5] Look smart!

---
 homeassistant/helpers/system_info.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/homeassistant/helpers/system_info.py b/homeassistant/helpers/system_info.py
index 4d161ae921f..1862ad54359 100644
--- a/homeassistant/helpers/system_info.py
+++ b/homeassistant/helpers/system_info.py
@@ -80,7 +80,7 @@ async def async_get_system_info(hass: HomeAssistant) -> dict[str, Any]:
 
     # Determine installation type on current data
     if info_object["docker"]:
-        if info_object["user"] == "root" and is_official_image():
+        if is_official_image():
             info_object["installation_type"] = "Home Assistant Container"
             info_object["container_arch"] = await async_get_container_arch(hass)
         else:
-- 
2.53.0

